src/webWorker/decodeTask.js

```diff
@@ -44,9 +49,13 @@ function handler(data, doneCallback) {
   // TODO switch to promise
   function finishedCallback(imageFrame) {
     if (!imageFrame.pixelData) {
-      throw new Error(
+      doneCallback(
```

Comment:
also here, why we are calling doneCallback when there is no pixelData, throw made sense a lot

Reply:
Throw isn't ever handled because this is execute asynchronously to the response code.

Reply:
I don't understand. Is there any catch block wrapping this function? Throwing error in async function is treated the same as sync function. If there is another catch block higher up in the code, it will be passed to that block for handling. If there is no catch block, the error will be logged to the console.